prov/util/src/util_mem_monitor.c

```diff
@@ -187,6 +187,26 @@ int ofi_monitors_add_cache(struct ofi_mem_monitor **monitors,
 	enum fi_hmem_iface iface;
 	struct ofi_mem_monitor *monitor;
 	unsigned int success_count = 0;
+	int iface_enabled[OFI_HMEM_MAX] = {
+		[FI_HMEM_SYSTEM] = 1,
+#ifdef HAVE_LIBCUDA
```

Comment:
Rather than using this as an ifdef, we should always define it as 0 or 1.  Then the code just needs:
```
[FI_HMEM_CUDA] = HAVE_LIBCUDA,
[FI_HMEM_ROCR] = HAVE_ROCR,
[FI_HMEM_ZE] = HAVE_ZE
```

Reply:
I agree. However, thanks to @rajachan's advise, we can get that information from one place. so this part is not needed any more.